jni/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+################################################################################################
+#
+# Xpedite script to compile Java files to create .jar files and .h files
+#
+# Author: Brooke Elizabeth Cantwell, Morgan Stanley
+#
+################################################################################################
+
+project(xpedite C CXX ASM)
+find_package(Java REQUIRED)
+find_package(JNI REQUIRED)
+include(UseJava)
+find_package(Java 1.8)
+
+execute_process(
+  COMMAND gradle xpediteJar
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+get_filename_component(SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR} DIRECTORY)
+set(XPEDITE_JNI_HEADER_PATH ${SRC_DIR}/install/jni)
+file(GLOB_RECURSE JAVA_FILES "src/main/com/xpedite/*.java")
+execute_process(
+  COMMAND javac -cp ${CMAKE_CURRENT_SOURCE_DIR}/jar/javassist.jar:${CMAKE_CURRENT_SOURCE_DIR}/src/main ${JAVA_FILES} -h ${XPEDITE_JNI_HEADER_PATH}
```

Comment:
Isn't this building the java files twice ? Shouldn't gradle generate the header file, when it generates the jar ?